File: Ensembles_Arithmétique/Code/is_prime.rs
fn is_prime(n: usize) -> bool{
    
    if n < 2 {
        return false;
    } 
    if n < 4 {
        return true;
    }
    if n%2 == 0 {
        return false;
    }

    let mut m: usize = 3;
    while m*m <= n {
        if n%m ==0 {
            return false;
        }
        m += 2;
    }

    true
}

File: Ensembles_Arithmétique/Code/isPrime.hs
-- test if an integer is divisible by another
isDivisible :: Integer -> Integer -> Bool
isDivisible a b = b * (div a b) == a

-- check that p is not divisible by an integer between q and the square root of p
testNonDivisible :: Integer -> Integer -> Bool
testNonDivisible p q | q*q > p = True
                     | isDivisible p q = False
                     | otherwise = testNonDivisible p (q+1)

-- check if a non-negative integer p is prime
isPrime :: Integer -> Bool
isPrime p | p <= 1 = False
          |otherwise = testNonDivisible p 2
